automerge-frontend/src/state_tree/resolved_path.rs

```diff
@@ -981,10 +981,11 @@ fn condense_insert_ops(ops: Vec<amp::Op>) -> Vec<amp::Op> {
                     cur_multiset_start = None;
                     new_ops.push(v);
                 }
-                (Some(typ), Some(scalar)) => match typ == discriminant(&scalar) {
-                    // there is a multiset in progress & the current op could be part of it
-                    true => cur_prim_vals.push(scalar),
-                    false => {
+                (Some(typ), Some(scalar)) => {
+                    if typ == discriminant(&scalar) && v.insert {
+                        // there is a multiset in progress & the current op could be part of it
+                        cur_prim_vals.push(scalar)
+                    } else {
```

Comment:
Should there be a case for the types being the same and it actually being a set, not an insert?

Reply:
I _think_ the logic is the same right? If the current discriminant is scalarvalue but the op is not `insert` then we can't do a multi insert, otherwise we can't. In the latter case it doesn't actually matter whether we're in a multi set or not. Unless you mean we could avoid doing `finish_multiset` in the case where we're not in a multiset?

Reply:
I was thinking that it might be different for a run of sets but I think this is fine.